GlotPress/GlotPress-WP
# Remove the glossary tooltip inside the HTML tags

## Problem

When GlotPress has an HTML tag in the glossary, it shows the glossary tooltip in the HTML tag.

![image](https://github.com/GlotPress/GlotPress/assets/1667814/b25054f5-402c-41d6-908d-409e8dc85458)

Fix https://github.com/GlotPress/GlotPress/issues/1382 and https://github.com/GlotPress/GlotPress/issues/1385.



## Solution

This PR excludes the glossary tooltips inside the HTML tags, splitting the original strings (singular and plural) to avoid adding the glossary tooltips to the HTML tags.

![image](https://github.com/GlotPress/GlotPress/assets/1667814/c719a4fd-dbb8-4f80-b430-c76ae316315b)



## Testing Instructions

This PR adds 2 tests:
- test_map_glossary_entries_to_translation_originals_excluding_terms_in_html_tags.
- test_map_glossary_entries_to_translation_originals_excluding_terms_in_html_tags_in_the_plural_origin.

To test it manually, you can:
1. Add an HTML tag (e.g., **strong**) to the glossary.
2. Add 2 new originals to be translated (singular and plural). E.g.:
  - Singular: `This is s strong test <strong>strong</strong>. This is another<dd>strong</dd>, very strong test with<img src="strong.img" /> images <hr/>.`
  - Plural: `Plural. This is s strong test <strong>strong</strong>. This is another<dd>strong</dd>, very strong test with<img src="strong.img" /> images <hr/>.`
4. Check that the tooltip for the `strong`word only appears outside the HTML tags.
